### PR TITLE
Refactor `MapboxSpeechApi`s to make them non-cancelling

### PR DESCRIPTION
**Pull request by Guardiola31337**
### Description

Refactors `MapboxSpeechApi`s to make them non-cancelling
Fixes https://github.com/mapbox/mapbox-navigation-android/issues/4403

### Changelog

```
<changelog>Refactored `MapboxSpeechApi`s to make them non-cancelling.</changelog>
```

Looked at `MapboxManeuverApi` and the changes brought in https://github.com/mapbox/mapbox-navigation-android/pull/4482 (concretely `getRoadShields`) and thought about what could be "ported" to `MapboxSpeechApi` and ended up with a really simple solution. We can remove the job restriction and problem solved. We don't need to wait for instructions at once (as opposed to shields). Would that be enough? Am I missing something? 🤔 

Opening this as draft for discussing ☝️

### CODE REVIEW

**Comment by abhishek1508** [comment]
If the requirement is to allow developers to `generate` multiple voice instructions at the same time; removing the `job.cancel()` and doing a `launch` every time `generate` is called should be enough.
My question is more on cancelling. If are allowing developers to generate multiple voice instructions simultaneously, should we allow them to decide which one to cancel. At the moment, calling `cancel` will cancel all the children jobs. Should we rather take an `id` and change the cancel api to `cancel(id)`?

cc @Guardiola31337

**Review by VitaminPSG — APPROVED** [review]

**Comment by Guardiola31337** [comment]
> My question is more on cancelling. If are allowing developers to generate multiple voice instructions simultaneously, should we allow them to decide which one to cancel. At the moment, calling `cancel` will cancel all the children jobs. Should we rather take an `id` and change the cancel api to `cancel(id)`?

Thought about this as well. @VitaminPSG @LukasPaczos what's your opinion on this? Do you think this is necessary as a client? What use case/s would this cover? 🤔

**Comment by VitaminPSG** [comment]
@Guardiola31337 @abhishek1508 Being able to cancel a specific instruction looks good from the API's point of view. However, for me this is not a mandatory case, usually the download is performed in parts (we preload 5 instructions at a time) and in the event of a route change, we can cancel all five instructions or leave them and put them in the cache.

**Comment by abhishek1508** [comment]
@Guardiola31337 Let me know once this is ready to review and I will approve

**Comment by LukasPaczos** [comment]
This simple solution looks good enough for me, for now. The `content manager` introduced for shields purposes is much more complex for a couple of reasons:
- it supports in-memory caching
- it deduplicates shields requests
- it groups together multiple requests (for shields for a list of maneuvers)

For now, voice API does not support caching, and just in-memory caching wouldn't be appropriate here - shields, when downloaded, are basically displayed for the whole duration of navigation and the files are much smaller, so it makes sense to keep them in memory. This is not true for voice instructions.

Being able to cancel requests independently is a nice-to-have, a bigger issue might be deduplicating of requests. But since a voice instruction is typically only played once, we have a smaller risk of redownloading an instruction than we have for shields, which are often duplicated.

I think we can easily address all of the above issues once we introduce a proper voice instruction cache, and integrate it with the `content manager` that shields use right now.

File: examples/src/main/java/com/mapbox/navigation/examples/core/MapboxVoiceActivity.kt [hunk 1]
```diff
@@ -270,7 +259,26 @@ class MapboxVoiceActivity : AppCompatActivity(), OnMapLongClickListener {
...
+                                "DEBUG ssmlAnnouncement: ${value.announcement.ssmlAnnouncement}"
+                            )
+                            Log.e(
+                                "DEBUG",
+                                "DEBUG ssmlAnnouncement file: ${value.announcement.file}"
+                            )
+                        }
+                    )
+                }
+            }
```
Comment (LukasPaczos) [inline examples/src/main/java/com/mapbox/navigation/examples/core/MapboxVoiceActivity.kt:281]: Good to revert?

File: examples/src/main/java/com/mapbox/navigation/examples/core/MapboxVoiceActivity.kt [hunk 2]
```diff
@@ -317,7 +325,16 @@ class MapboxVoiceActivity : AppCompatActivity(), OnMapLongClickListener {
...
+                    routes[0].legs()?.get(0)?.steps()?.take(5)?.forEach {
+                        it.voiceInstructions()?.let { voiceInstructions ->
+                            voiceInstructionsCache.addAll(voiceInstructions)
+                        }
+                    }
+                    voiceInstructionsCache.forEach {
+                        Log.e("DEBUG", "DEBUG cache ssmlAnnouncement: ${it.ssmlAnnouncement()}")
+                    }
+                    // TODO commented out for testing purposes
+                    // mapboxNavigation.setRoutes(routes)
```
Comment (LukasPaczos) [inline examples/src/main/java/com/mapbox/navigation/examples/core/MapboxVoiceActivity.kt:337]: Good to revert?

File: examples/src/main/java/com/mapbox/navigation/examples/core/MapboxVoiceActivity.kt (hunk 1 above)
Comment (Guardiola31337) [reply examples/src/main/java/com/mapbox/navigation/examples/core/MapboxVoiceActivity.kt:281]: Yeah I was on it while fixing the conflict with `main` 😅

File: examples/src/main/java/com/mapbox/navigation/examples/core/MapboxVoiceActivity.kt (hunk 2 above)
Comment (Guardiola31337) [reply examples/src/main/java/com/mapbox/navigation/examples/core/MapboxVoiceActivity.kt:337]: Same as above 😅

**Review by LukasPaczos — APPROVED** [review]
LGTM

**Pull request merged**